frontend/packages/knative-plugin/src/components/add/EventSourceForm.tsx

```diff
@@ -23,32 +25,50 @@ const EventSourceForm: React.FC<FormikProps<FormikValues> & OwnProps> = ({
   errors,
   handleSubmit,
   handleReset,
+  setFieldValue,
+  setFieldTouched,
   status,
   isSubmitting,
   dirty,
+  validateForm,
   namespace,
   projects,
-}) => (
-  <Form className="co-deploy-image" onSubmit={handleSubmit}>
-    <EventSourcesSelector eventSourceList={useEventSourceList(namespace)} />
-    {values.type === EventSources.CronJobSource && <CronJobSection />}
-    {values.type === EventSources.SinkBinding && <SinkBindingSection />}
-    {values.type === EventSources.ApiServerSource && <ApiServerSection namespace={namespace} />}
-    <SinkSection namespace={namespace} />
-    <AppSection
-      project={values.project}
-      noProjectsAvailable={projects?.loaded && _.isEmpty(projects.data)}
-    />
-    <FormFooter
-      handleReset={handleReset}
-      errorMessage={status && status.submitError}
-      isSubmitting={isSubmitting}
-      submitLabel="Create"
-      sticky
-      disableSubmit={!dirty || !_.isEmpty(errors)}
-      resetLabel="Cancel"
-    />
-  </Form>
-);
+}) => {
+  const eventSourceModels = eventSourceData?.models;
+  const handleChange = (item: string) => {
```

Comment:
Can we have this in `EventSource` file and send it as a prop to the form component. The name can be something like `handleEventSourceChange`. Or else why don't you add  it in the `EventSourceSelector` itself? For every event source we will need to have the `apiVersion` so it makes sense to set it from within the `EventSourceSelector`. According to me the form component should only define the structure of the form not be concerned with any other logic.

Reply:
Have moved here as yes now it's `ApiServer` but later could be used for setting `yamlData` as well based on Type.

Reply:
added in `EventSourceSelector` itself